server/tests/common/backend.py

```diff
@@ -68,6 +69,35 @@ def app(backend: Backend, monkeypatch: pytest.MonkeyPatch) -> Iterator[AsgiApp]:
     asgi_app.state.backend = backend
     yield asgi_app
 
+@contextmanager
+def run_asgi_server(app: AsgiApp) -> Iterator[tuple[str, int]]:
+    from multiprocessing import Process
+
+    from parsec.asgi import serve_parsec_asgi_app
+
+    def get_local_port(host: str) -> int:
```

Comment:
Add a TODO comment:
```
# TODO: This is a hack to get a unique-enough port for the server.
#              It would be cleaner to pass port=0, then parse the sub-process
#              stdout to retrieve the actual port in use.
```

Reply:
done